product-scenarios/2-Collaborative-API-Development/2.1-Manage-publisher-portal-operations-for-users/2.1.1-Manage-API-publishing-according-to-the-publisher-roles-and-permission/src/test/java/org/wso2/am/scenario/tests/rest/api/publisherRoleAndPermission/PublishAPIByValidRoleCategoryTestCase.java

```diff
@@ -0,0 +1,250 @@
+/*
+ * Copyright (c) 2018, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.wso2.am.scenario.tests.rest.api.publisherRoleAndPermission;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+import org.wso2.am.scenario.test.common.APIPublisherRestClient;
+import org.wso2.am.scenario.test.common.ScenarioDataProvider;
+import org.wso2.am.scenario.test.common.ScenarioTestBase;
+import org.wso2.carbon.automation.test.utils.http.client.HttpResponse;
+import org.wso2.am.integration.test.utils.bean.APILifeCycleState;
+import org.wso2.am.integration.test.utils.bean.APILifeCycleStateRequest;
+import org.wso2.am.scenario.test.common.APIRequest;
+
+import java.net.URL;
+import java.util.Properties;
+import javax.ws.rs.core.Response;
+
+import static org.testng.Assert.assertEquals;
+
+import java.util.*;
+
+import static org.testng.Assert.assertTrue;
+import static org.testng.Assert.assertFalse;
+
+public class PublishAPIByValidRoleCategoryTestCase extends ScenarioTestBase {
+
+    private APIPublisherRestClient apiPublisher;
+    private String publisherURL;
+    private String storeURL;
+    private APIRequest apiRequest;
+    private Properties infraProperties;
+
+    private String apiName;
+    private String apiContext = "/verify";
+    private String apiResource = "/find";
+    private String apiVisibility = "public";
+    private String apiVersion = "1.0.0";
+    private String tierCollection = "Gold,Bronze";
+    private String backendEndPoint = "http://ws.cdyne.com/phoneverify/phoneverify.asmx";
+    private String creatorUser;
+    private String testUser;
+    private String password;
+    private int count = 0;
+
+    private static final Log log = LogFactory.getLog(PublishAPIByValidRoleCategoryTestCase.class);
+    private final String ADMIN_LOGIN_USERNAME = "admin";
+    private final String ADMIN_PASSWORD = "admin";
+    Map<String, String> apiNames = new HashMap<>();
+    List<String> userList = new ArrayList();
+    List<String> roleList = new ArrayList();
+
+    String userRole;
+
+    @BeforeClass(alwaysRun = true)
+    public void init() {
+
```

Comment:
we do not need the following logic, as this is being executed in parent constrcutor